firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java

```diff
@@ -85,14 +88,29 @@
       FirebaseApp app,
       Context context,
       String projectId,
-      String region,
+      String regionOrCustomDomain,
       ContextProvider contextProvider) {
     this.app = app;
     this.client = new OkHttpClient();
     this.serializer = new Serializer();
     this.contextProvider = Preconditions.checkNotNull(contextProvider);
     this.projectId = Preconditions.checkNotNull(projectId);
-    this.region = Preconditions.checkNotNull(region);
+
+    boolean isRegion;
+    try {
+      new URL(regionOrCustomDomain);
+      isRegion = false;
+    } catch (MalformedURLException malformedURLException) {
+      isRegion = true;
+    }
+
+    if (isRegion) {
```

Comment:
Nit: you could just move this logic inside of the `try/catch` and not need `isRegion` at all.  But totally fine either way.

Reply:
Since `region` and `customDomain` are final they can't be set both in the try and the catch.

I'm gussing that Java considers the whole try and not just the lines that could cause the exception.

Reply:
Ah interesting, that's fair then!